processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -268,6 +252,24 @@ public File merge(
     );
   }
 
+  private static List<IndexableAdapter> toIndexableAdapters(List<QueryableIndex> indexes)
+  {
+    // We are materializing the list for performance reasons. Lists.transform
+    // only creates a "view" of the original list, meaning the function gets
+    // applied every time you access an element.
+    return Lists.transform(
```

Comment:
The behavior is different here. The list used to be materialized (`Lists.newArrayList`) but now it's a view (`Lists.transform`). The comment says it should be materialized, so I think the old code was right.